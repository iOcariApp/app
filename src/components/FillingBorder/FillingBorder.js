import React from "react";
import PropTypes from "prop-types";
import { Animated } from "react-native";
import styles from "./FillingBorder.style";

class FillingBorder extends React.Component {
  state = {
    scale: new Animated.Value(0),
  };

  componentDidMount = () => {
    Animated.timing(this.state.scale, {
      toValue: 1,
      duration: this.props.duration,
    }).start();
  };

  render = () => {
    const { scale } = this.state;
    return (
      <Animated.View
        style={[
          styles.main,
          {
            transform: [
              {
                scaleX: scale,
              },
            ],
          },
        ]}
      />
    );
  };
}

FillingBorder.propTypes = {
  duration: PropTypes.number,
};
FillingBorder.defaultProps = {
  duration: 300,
};

export default FillingBorder;
