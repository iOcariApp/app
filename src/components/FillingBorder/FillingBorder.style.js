import { StyleSheet } from "react-native";
import { colors } from "theme";

export default StyleSheet.create({
  main: {
    width: "100%",
    height: 4,
    backgroundColor: colors.green,
  },
});
