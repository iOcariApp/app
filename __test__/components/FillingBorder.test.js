import React from "react";

import FillingBorder from "components/FillingBorder";

test("renders correctly", () => {
  testSnapshot(<FillingBorder />);
});

test("applies `color` prop", () => {
  testSnapshot(<FillingBorder color="black" />);
});
