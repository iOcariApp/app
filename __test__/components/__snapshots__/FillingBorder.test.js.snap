// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies \`color\` prop 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "height": 4,
      "transform": Array [
        Object {
          "scaleX": 1,
        },
      ],
      "width": "100%",
    }
  }
/>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#4DB6AC",
      "height": 4,
      "transform": Array [
        Object {
          "scaleX": 1,
        },
      ],
      "width": "100%",
    }
  }
/>
`;
